lib/iterator.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+class Iterator {
+  constructor(base) {
+    if (base[Symbol.iterator]) {
+      this.base = base[Symbol.iterator]();
+    } else if (base.next) {
+      this.base = base;
+    } else {
+      throw new TypeError('Base is neither Iterator nor Iterable');
+    }
+  }
+
+  [Symbol.iterator]() {
+    return this;
+  }
+
+  next() {
+    return this.base.next();
+  }
+
+  forEach(fn, thisArg) {
+    for (const value of this) {
+      fn.call(thisArg, value);
+    }
+  }
+
+  map(mapper, thisArg) {
+    // eslint-disable-next-line no-use-before-define
+    return new MapIterator(this, mapper, thisArg);
+  }
+
+  filter(predicate, thisArg) {
+    // eslint-disable-next-line no-use-before-define
+    return new FilterIterator(this, predicate, thisArg);
+  }
+
+  each(predicate, thisArg) {
+    for (const value of this) {
+      if (!predicate.call(thisArg, value)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  find(predicate, thisArg) {
+    for (const value of this) {
+      if (predicate.call(thisArg, value)) {
+        return value;
+      }
+    }
+  }
+
+  includes(element) {
```

Comment:
Could we add [`fromIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Parameters)?

Reply:
I'd like :+1:

Reply:
An index doesn't make sense for `Map`, `Set` or `Iterable` in general for what it is worth.

Reply:
That's true, the what do you think about .take and .skip methods?

Reply:
Implemented `skip(amount)` and `take(amount)`.